### ISSUE TITLE
Change command + `=` syntax

### ISSUE THREAD

**Issue by hlaaftana** [nim-lang/RFCs#442]
Currently this:

```nim
foo a = b
```

parses as this:

```nim
foo(a) = b
```

The proposal is to make this mean:

```nim
foo(a = b)
```

and maybe even allow `foo a = b, c = d` syntax.

Reasoning:

* I can't think of a justification for the current syntax. `foo(a) = b` or `a.foo = b` are already possible and not much less ergonomic.
* Other languages with command syntax, such as Ruby, already employ this proposed syntax, which just makes Nim's current syntax weirder.
* I've never seen the current syntax in code likely because of the above problems.
* Named arguments in Nim are used like `foo(a = b, c = d)`, so there is a clear use for this syntax in command form. Macros and DSLs would also benefit from this.

This change would definitely break code which uses the current syntax. However I genuinely have never seen *any* code which does. And it can be easily tied to a big, backwards-incompatible release like 2.0.

**Comment by mratsim** [comment]
That would break the rule "unary binds stronger than binary". It's an easy enough rule to remember, I'm not sure the exception is worth it. And later all parsing/syntax highlighting packages need to be updated.

**Comment by hlaaftana** [comment]
> That would break the rule "unary binds stronger than binary".

But that rule already doesn't apply for `a b + c`, or even `a b += c` or `a b := c`. `=` is the exception here.

> And later all parsing/syntax highlighting packages need to be updated.

I doubt they parse this correctly already.

**Comment by Araq** [comment]
I like the idea.

**Comment by c-blake** [comment]
It may be worth observing here that named arguments already work for non-first params:
```Nim
proc foo(x: int, a=2, b=3) = discard
foo 1, b=2, a=3
```

**Comment by metagn** [comment]
Was mentioned but has been included more thoroughly in the proposal.

**Comment by c-blake** [comment]
I agree "all arguments working" is a good consistency principle.  It took me a while to discover that it worked for non-first args, actually.  Sorry if I didn't notice the mention already.  It may be the strongest pro-argument here, but I think it would be good for other Nimmers to weigh in.

You may also be interested in what I call ["named do"](https://forum.nim-lang.org/t/8110#53255) which I think of as closely related - making named parameter passing work in "code block notation" as well as "code expression notation" (which already works).  You just need a way to connect the parameter name and `foo.do:` or `do(foo):` seem natural fits.
